napari/_tests/test_windowsettings.py

```diff
@@ -1,8 +1,28 @@
+from qtpy.QtCore import QRect
+
 from napari.settings import get_settings
 
 
-def test_singlescreen_window_settings(make_napari_viewer):
+class ScreenMock:
+    def __init__(self):
+        self._geometry = QRect(0, 0, 1000, 1000)
+
+    def geometry(self):
+        return self._geometry
+
+
+def screen_at(point):
+    if point.x() < 0 or point.y() < 0 or point.x() > 1000 or point.y() > 1000:
+        return None
+    return ScreenMock()
+
+
+def test_singlescreen_window_settings(make_napari_viewer, monkeypatch):
     """Test whether valid screen position is returned even after disconnected secondary screen."""
+
+    monkeypatch.setattr(
+        "napari._qt.qt_main_window.QApplication.screenAt", screen_at
+    )
```

Comment:
This is the fix for running a test locally on my machine. I could extract it to separate PR. But I need this for local testing as `(0,0)` is outside my screens:

![obraz](https://github.com/napari/napari/assets/3826210/7ab35aee-0fb1-495f-bbef-9ff8df3adcdb)

Reply:
LOL that's funny. Yeah I think it might not make sense to have it in main and certainly not to have it in this PR...

Reply:
> I think it might not make sense to have it in main

Why? I think that we should go in the direction that allows run all test locally.

Reply:
Yeah but very few people have a geometry where (0, 0) is outside it? But, sure, maybe it can be discussed more extensively in a separate PR (and it would make the motivation for the code easier to discover in the future).

Reply:
I have opened #6422